src/api/controllers/lines.js

```diff
@@ -342,16 +342,27 @@ function updateLineStatus(req, res) {
                 });
             } else {
                 try {
-                    var msg = await line.updateStatus(req.body.status);
-                    res.status(200).json({
-                          "message": "Successfully updated line with status " + req.body.status
-                    });
-                } catch (err) {
-                    console.log(err);
-                    res.status(400).json({
-                        "message": err
-                    });
-                }
+                    if (req.body.status === 'startrecruiter' || req.body.status === 'finishrecruiter' && (user.user_type !== 'recruiter' || user.user_type !== 'admin')) {
```

Comment:
im pretty sure this should be `user.user_type !== 'recruiter' && user.user_type !== 'admin'`, cause its recruiters or admins that can pass, so by demorgans if u wanna catch the opposite its -A V -B.

Reply:
please test this thoroughly with postman, you should create a recruiter and admin account via /register and copy the token into your env. A user can be a recruiter or an admin and be able to update the line to startrecruiter or finishrecruiter

Reply:
this might need to be req.user.user_type, not just user.user_type

Reply:
consider using a switch case statement to separate the checks by the passed in status. this would make the code much more readable

there is an example of this in models/lines.js:

```
    switch (status) {

        case 'notification':
            if (this.status !== 'preline') {
                msg = "LineUpdateError: Cannot move to notification status from status " + this.status;
            }
            break;
        case 'inline':
            if (this.status !== 'notification' && this.status !== 'preline') {
                msg = "LineUpdateError: Cannot move to inline status from status " + this.status;
            }
            break;
        case 'startrecruiter':
            if (this.status !== 'inline') {
                msg = "LineUpdateError: Cannot move to startrecruiter status from status " + this.status;
            }
            break;
        case 'finishrecruiter':
            if (this.status !== 'startrecruiter') {
                msg = "LineUpdateError: Cannot move to finishrecruiter status from status " + this.status;
            }
            break;
        case 'timeoutchurn':
            if (this.status !== 'notification' && this.status !== 'inline' && this.status !== 'startrecruiter') {
                msg = "LineUpdateError: Cannot time out when in status " + this.status;
            }
            break;
        case 'preline':
            msg = "Cannot change status to preline - exit and reenter this line";
            break;
        case 'voluntarychurn':
            break; //can enter or exit at any time
        default:
            msg = "Invalid status desired: " + this.status;
            break;
    }
```